File: src/features/typing/data/ts/search_a_2d_matrix_ii.ts
function searchMatrix(matrix: number[][], target: number): boolean {
    let i = 0;
    let j = matrix[0].length - 1;
    while (i < matrix.length && j >= 0) {
        if (matrix[i][j] === target) {
            return true;
        } else if (matrix[i][j] < target) {
            i++;
        } else {
            j--;
        }
    }
    return false;
}

File: src/features/typing/data/go/search_a_2d_matrix_ii.go
package main

func searchMatrix(matrix [][]int, target int) bool {
	i, j := 0, len(matrix[0])-1
	for i < len(matrix) && j >= 0 {
		switch {
		case matrix[i][j] == target:
			return true
		case matrix[i][j] < target:
			i++
		default:
			j--
		}
	}
	return false
}
